Feature/299 implement dark mode

dbaslan opened the pull request:
This branch implements dark mode to the web app, in line with our stretch goal. 😎 Please let me know if I've missed any text or buttons, otherwise awaiting approval. The implementation is a bit rudimentary, all it does is darken the background image and the header color, and turn the headers and text white where necessary. It does not make use of Material UI's built-in dark mode, and I had to use inline CSS on a bunch of elements to get it done. It would be too chaotic otherwise. See #299

dbaslan commented:
Here are some screenshots:

![Ekran görüntüsü 2023-12-20 200444](https://github.com/SWE574-Fall2023-Group1/SWE574-Fall2023-G1/assets/66262500/7a6dfc07-aeb8-428a-9ddf-1e56f776e889)

![Ekran görüntüsü 2023-12-20 200422](https://github.com/SWE574-Fall2023-Group1/SWE574-Fall2023-G1/assets/66262500/7cee1d54-8db7-4048-b7cf-724f79532df7)

ckertam commented:
I need to test it more after I test it I will make my comment

sadikkuzu commented on backend/frontend/src/App.js:
```diff
@@ -13,11 +13,33 @@ const theme = createTheme({
+const customTheme = createTheme({
+  palette: {
+    primary: {
+      main: '#ff9800', // Change this to your custom color
```
```suggestion
      main: '#ff9800',
```

Let's remove chatgpt comments;)

dbaslan replied:
Oops, co-pilot did me dirty 😅 Removing

sadikkuzu commented on backend/frontend/src/App.js:
```diff
@@ -13,11 +13,33 @@ const theme = createTheme({
+      main: '#ff9800', // Change this to your custom color
+    },
+    secondary: {
+      main: '#f57c00', // Change this to your custom color
```
```suggestion
      main: '#f57c00',
```

Let's remove chatgpt comments;)

sadikkuzu commented on backend/frontend/src/App.js:
```diff
@@ -13,11 +13,33 @@ const theme = createTheme({
 function App() {
+  const [currentTheme, setCurrentTheme] = useState('default'); // add a state variable for the current theme
+
+  const toggleTheme = () => { // add a function to toggle the theme
```
```suggestion
  const toggleTheme = () => {
```

Let's remove chatgpt comments;)

sadikkuzu commented on backend/frontend/src/App.js:
```diff
@@ -13,11 +13,33 @@ const theme = createTheme({
 
+const customTheme = createTheme({
+  palette: {
+    primary: {
```
Just asking
We have `default` and `custom` themes in the code.
Do we use these `primary` and `secondary` definitions later?

dbaslan replied:
I thought we did, but turns out we don't. Removing

sadikkuzu commented on backend/frontend/src/App.js:
```diff
@@ -13,11 +13,33 @@ const theme = createTheme({
+  const [currentTheme, setCurrentTheme] = useState('default'); // add a state variable for the current theme
+
+  const toggleTheme = () => { // add a function to toggle the theme
+    setCurrentTheme(currentTheme === 'default' ? 'custom' : 'default');
```
I prefer
default -> light
custom -> dark
naming

dbaslan replied:
I did this on purpose, because using the word dark triggers Material UI's built-in dark mode, which messes up the whole app. If you'd like, I can do defaultLight and customDark? @sadikkuzu

sadikkuzu replied:
Oh ok, i see. That makes sense. Thanks for clarification 🙏🏼

Well, current default&custom naming is ok 👍🏼 no need to change
Thanks again 🙏🏼✨🚀

sadikkuzu reviewed (changes requested):
I have some comments and tiny little change-requests.
Thanks in advance.

sadikkuzu reviewed (commented):
Previous/Next buttons

I cannot see button labels in dark mode:

light:
![light](https://github.com/SWE574-Fall2023-Group1/SWE574-Fall2023-G1/assets/23168063/6a389cb5-1b1b-4612-9af7-dbdc14b1d2ea)

dark:
![dark](https://github.com/SWE574-Fall2023-Group1/SWE574-Fall2023-G1/assets/23168063/494fe153-9a69-4439-a26f-ef094f2affa2)

dbaslan commented:
Tried a bunch of solutions for the buttons, nothing worked. In the end I made the buttons disappear when disabled (i.e. when there is no previous or next page). Changed the dark mode button text color as per @sadikkuzu's request and fixed some minor visibility issues as well.

sadikkuzu reviewed (commented):
Create Memory page (Include time):

* Normal Date
* Interval Date

![create-memory](https://github.com/SWE574-Fall2023-Group1/SWE574-Fall2023-G1/assets/23168063/b87d4348-7e14-4e43-a749-36566e9ae7a6)

sadikkuzu reviewed (commented):
Create Memory page (location alias):

![location-alias](https://github.com/SWE574-Fall2023-Group1/SWE574-Fall2023-G1/assets/23168063/2d650989-2734-4664-82ec-3fdc171da570)

![text-selected-version](https://github.com/SWE574-Fall2023-Group1/SWE574-Fall2023-G1/assets/23168063/a13af433-a004-4f66-a453-d3a506a3b082)

sadikkuzu commented:
> Tried a bunch of solutions for the buttons, nothing worked. In the end I made the buttons disappear when disabled (i.e. when there is no previous or next page). Changed the dark mode button text color as per @sadikkuzu's request and fixed some minor visibility issues as well.

Cool solution 👍🏼 

![next](https://github.com/SWE574-Fall2023-Group1/SWE574-Fall2023-G1/assets/23168063/55b5c988-765d-48d4-aa78-57b6c10ab251)

![previous](https://github.com/SWE574-Fall2023-Group1/SWE574-Fall2023-G1/assets/23168063/b132c023-24e1-48f6-b5d1-922d0fc679da)

dbaslan commented:
Done and done @sadikkuzu 👍

ckertam reviewed (approved):
LGTM! It was our dream I guess 😅

sadikkuzu reviewed (approved):
Perfecto 👌🏼
